### ISSUE TITLE
[QUESTION] ContextMapper in the context of .NET?

### ISSUE THREAD

**Issue by CesarD** [ContextMapper/context-mapper-dsl#363]
Sorry if this is out of place, but I figured I ask this in this repo as it seems to me the main one for ContextMapper: I see that there're a lot of ContextMapper's tools for working in the context of Java projects, like the ArchUnit extension, the context map discovery and some other; and since I work mainly on the .NET platform I can't help but wonder if there's any way to leverage ContextMapper over there as well.

Not asking for you guys to develop something for .NET specifically, but mainly if anybody knows of some existing tooling that already leverages ContextMapper (or some of its artifacts) for .NET.

**Comment by stefan-ka** [comment]
Questions are always welcome @CesarD; as well as contributions ;)

I am not aware of similar tools, like the ArchUnit extension or the reverse engineering library, for .NET. But contributions are very welcome of course, as already said. I assume it should somehow be possible to write something similar for .NET. The main question would probably be whether you write your own CML parser with .NET technologies, or whether you expect the users to have Java installed to use our Xtext stack (standalone library); and how the interface would be implemented to call our Java library.

**Comment by CesarD** [comment]
Yeah, the reusing of the already existing libraries would be ideal, but I'm not sure how we would be able to interface to them in order to reuse them, so that would leave the parser option. Do you reckon if something like ANTLR would help? I know it works on both platforms (and more)...

**Comment by socadk** [comment]
Hi @CesarD not directly related to Context Mapper, but about DDD in a .NET context: there is an [nMolecules](https://xmolecules.org/) project with implementations in Java and .NET: https://github.com/xmolecules/nmolecules
